source/src/main/groovy/com/kezong/fataar/FlavorArtifact.groovy

```diff
@@ -165,7 +165,7 @@ class FlavorArtifact {
                         && toFlavor == subFlavor.name
                         && variant.buildType.name == subVariant.buildType.name) {
                     try {
-                        bundleTaskProvider = VersionAdapter.getBundleTaskProvider(project, subVariant)
+                        bundleTaskProvider = VersionAdapter.getBundleTaskProvider(project, variant)
```

Comment:
Subvariant’s name is not equal to variant's name，What is needed here is a subvariant, not a variant.
Maybe you should change the parameters of `VersionAdapter.getBundleTaskProvider`

Reply:
This line is inside condition `if (variant.name == subVariant.name)`. So, in current 
Yeah, type of subVariant became more complex. I'll try to find correct interface later.

Maybe, `VersionAdapter.getBundleTaskProvider` should receive name directly?

Reply:
The first reference is inside `if (variant.name == subVariant.name)`, but the second is not.

You can change the `VersionAdapter.getBundleTaskProvider` to receive the variant's name directly.